### PR TITLE
Benchmark ips formatting improvements 

### PR DESCRIPTION
**Pull request by will**
Two improvements, use relative stddev instead and also use SI suffixes on the means. The relative stddev I think is easier to understand. The shorter means is because in many of the cases I've seen the IPS is so high, it's hard to read the numbers.

now
```
stdlib 2054429.34 (± 142437.52)       fastest
reimpl 1952512.39 (± 208080.29)  1.05× slower
```
after
```
stdlib   2.06M (± 6.54%)       fastest
reimpl   1.95M (±11.47%)  1.04× slower  
```

### CODE REVIEW

**Comment by jhass** [comment]
LGTM though maybe we can get 3-5 or possibly even all significant digits? The difference between `20.05M` and `20.5M` is still interesting and it might look odd if it claims one is faster while displaying `20M` for both.

**Comment by ysbaddaden** [comment]
Just saying: 2054429.34 => 2_054_429.34 => 2 millions, not 20 ...

Also, maybe it's because I'm a frenchy, but it took me some seconds to understand `M` as million. But I like the +/- becoming a percentage, that's easier to read.

**Comment by will** [comment]
Updated the mean to add more sig figs, and fixed the incorrect division, and added specs. Thanks @jhass and @ysbaddaden 

> Also, maybe it's because I'm a frenchy, but it took me some seconds to understand M as million.

Do you not use MB for million bytes in France? (pedantry differences between MB and MiB notwithstanding)

**Comment by ysbaddaden** [comment]
We write "Mo" for "mégaoctets" as a derivate from megabytes. We usually don't use the K or M for thousands and millions.

Anyway, it's fine.

File: src/benchmark/benchmark.cr [hunk 1]
```diff
@@ -12,11 +12,12 @@ require "./**"
...
 #
-# This generates the following output:
+# This generates the following output showing the mean iterations per second,
+# the standard deviation relative to the mean, and a comparison:
 #
 # ```text
-#   short sleep    91.82 (±  1.11)  8.72× slower
-# shorter sleep   800.98 (±  4.72)       fastest
+#   short sleep    91.82 (± 2.51%)  8.72× slower
+# shorter sleep   800.98 (± 1.10&)       fastest
```
Comment (ysbaddaden) [inline src/benchmark/benchmark.cr]: Small typo: & instead of %

File: src/benchmark/ips.cr [hunk 2]
```diff
@@ -153,6 +156,21 @@ module Benchmark
...
+      def human_mean
+        pair = case Math.log10(mean)
+               when 0..3
+                 {mean, ' '}
+               when 3..6
+                 {mean/1_000, 'k'}
+               when 6..9
+                 {mean/1_000_000, 'M'}
+               else
+                 {mean/1_000_000_000, 'G'}
```
Comment (ysbaddaden) [inline src/benchmark/ips.cr]: Isn't MM used for billions?

Comment (will) [inline src/benchmark/ips.cr]: Usually I see the MM only when talking of money. I was basing the letters and their capitalization from https://en.wikipedia.org/wiki/Metric_prefix#List_of_SI_prefixes

File: src/benchmark/benchmark.cr (hunk 1 above)
Comment (will) [inline src/benchmark/benchmark.cr]: Thanks, fixed

**Comment by jhass** [comment]
Thank you!

**Pull request merged**